skyportal/handlers/api/enum_types.py

```diff
@@ -0,0 +1,60 @@
+from baselayer.app.access import auth_or_token
+
+from ..base import BaseHandler
+from ...enum_types import (
+    ALLOWED_SPECTRUM_TYPES,
+    ALLOWED_MAGSYSTEMS,
+    ALLOWED_BANDPASSES,
+    THUMBNAIL_TYPES,
+    FOLLOWUP_PRIORITIES,
+    ALLOWED_API_CLASSNAMES,
+)
+
+
+class EnumTypesHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
+        ---
+        description: Retrieve enum types in the DB
+        tags:
+          - system_info
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            ALLOWED_SPECTRUM_TYPES:
+                              type: array
+                              description: list of allowed spectrum types
+                            ALLOWED_MAGSYSTEMS:
+                              type: array
+                              description: list of allowed magnitude systems
+                            ALLOWED_BANDPASSES:
+                              type: array
+                              description: list of allowed bandpasses
+                            THUMBNAIL_TYPES:
+                              type: array
+                              description: list of allowed thumbnail types
+                            FOLLOWUP_PRIORITIES:
+                              type: array
+                              description: list of allowed followup priorities
+                            ALLOWED_API_CLASSNAMES:
+                              type: array
+                              description: list of allowed API classnames
+        """
+        data = {}
+        data["ALLOWED_SPECTRUM_TYPES"] = sorted(ALLOWED_SPECTRUM_TYPES)
+        data["ALLOWED_MAGSYSTEMS"] = sorted(ALLOWED_MAGSYSTEMS)
+        data["ALLOWED_BANDPASSES"] = sorted(ALLOWED_BANDPASSES)
+        data["THUMBNAIL_TYPES"] = sorted(THUMBNAIL_TYPES)
+        data["FOLLOWUP_PRIORITIES"] = sorted(FOLLOWUP_PRIORITIES)
+        data["ALLOWED_API_CLASSNAMES"] = sorted(ALLOWED_API_CLASSNAMES)
```

Comment:
Maybe the list of available enums should live in `enum_types` where they are defined? perhaps even in the `__all__` field?

Reply:
sqla_enum_types = [
    allowed_spectrum_types,
    allowed_bandpasses,
    thumbnail_types,
    instrument_types,
    followup_priorities,
    api_classnames,
    listener_classnames,
]

is defined, but those all have sa.Enum calls.

Reply:
I'm just thinking of how to make it easy to add enums in the future with the minimal number of places we'd have to go to make that change. We can make an enum of all enums! (please don't actually do this...)

Reply:
I think a future PR should rethink whether we actually want any enums at all, given how bad migrations are.